watertap/tools/analysis_tools/loop_tool/loop_tool.py

```diff
@@ -402,9 +402,18 @@ def get_diff_params(self, key, loop_value):
             force_rerun,
         )
 
+    def get_num_samples(self, value_dict):
+        num_samples = value_dict.get("num_samples")
+        if value_dict.get("num_samples") is not None:
+            return value_dict.get("num_samples")
+        elif value_dict.get("array") is not None:
+            return len(value_dict.get("array"))
+        else:
+            raise "Provide a valid sweep type, refer to parameter_sweep_reader for valid types"
```

Comment:
```suggestion
        num_samples = value_dict.get("num_samples")
        array = value_dict.get("array")
        if num_samples is not None and array is not None:
            raise RuntimeError("Supply `num_samples` or `array`, not both")
        if num_samples is not None:
            return num_samples
        if array is not None:
            return len(array)
        raise RuntimeError("Provide a valid sweep type, refer to parameter_sweep_reader for valid types")
```
I've made this method catch another edge case to provide a more robust implementation.